src/util.rs

```diff
@@ -7,149 +7,44 @@
-
-    // Make sure input fits in output.
-    if (SIZEOF_USIZE as u32 - input.leading_zeros() / 8) > L::U32 {
-        return Err(Error::SerializationError);
-    }
-
-    let mut output = GenericArray::default();
-    output[L::USIZE.saturating_sub(SIZEOF_USIZE)..]
-        .copy_from_slice(&input.to_be_bytes()[SIZEOF_USIZE.saturating_sub(L::USIZE)..]);
-    Ok(output)
-}
-
-/// Computes `I2OSP(len(input), max_bytes) || input` and helps hold output
-/// without allocation.
-pub(crate) struct Serialize<'a, L1: ArrayLength<u8>, L2: ArrayLength<u8> = U0> {
-    octet: GenericArray<u8, L1>,
-    input: Input<'a, L2>,
-}
-
-enum Input<'a, L: ArrayLength<u8>> {
-    Owned(GenericArray<u8, L>),
-    Borrowed(&'a [u8]),
-}
-
-impl<'a, L1: ArrayLength<u8>, L2: ArrayLength<u8>> IntoIterator for &'a Serialize<'a, L1, L2> {
-    type Item = &'a [u8];
-
-    type IntoIter = IntoIter<&'a [u8], 2>;
-
-    fn into_iter(self) -> Self::IntoIter {
-        // MSRV: array `into_iter` isn't available in 1.51
-        #[allow(deprecated)]
-        IntoIter::new([
-            &self.octet,
-            match self.input {
-                Input::Owned(ref bytes) => bytes,
-                Input::Borrowed(bytes) => bytes,
-            },
-        ])
-    }
-}
-
-impl<'a, L1: ArrayLength<u8>, L2: ArrayLength<u8>> Serialize<'a, L1, L2> {
-    // Variation of `serialize` that takes a borrowed `input.
-    pub(crate) fn from(input: &[u8]) -> Result<Serialize<L1>> {
-        Ok(Serialize {
-            octet: i2osp::<L1>(input.len())?,
-            input: Input::Borrowed(input),
-        })
-    }
-
-    pub(crate) fn from_owned(input: GenericArray<u8, L2>) -> Result<Serialize<'static, L1, L2>> {
-        Ok(Serialize {
-            octet: i2osp::<L1>(input.len())?,
-            input: Input::Owned(input),
-        })
-    }
-}
-
-macro_rules! chain_name {
-    ($var:ident, $mod:ident) => {
-        $mod
-    };
-    ($var:ident) => {
-        $var
-    };
-}
-
-macro_rules! chain_skip {
-    ($var:ident, $feed:expr) => {
-        $feed
-    };
-    ($var:ident) => {
-        &$var
-    };
+pub(crate) fn i2osp_2(input: usize) -> Result<GenericArray<u8, U2>> {
+    u16::try_from(input)
+        .map(|input| input.to_be_bytes().into())
+        .map_err(|_| Error::SerializationError)
 }
 
-/// The purpose of this macro is to replace
-/// [`concat`](alloc::slice::Concat::concat)ing slices into an [`Iterator`] to
-/// avoid allocation
-macro_rules! chain {
-    (
-        $var:ident,
-        $item1:expr $(=> |$mod1:ident| $feed1:expr)?,
-        $($item2:expr $(=> |$mod2:ident| $feed2:expr)?),+$(,)?
-    ) => {
-        let chain_name!(__temp$(, $mod1)?) = $item1;
-        let $var = (chain_skip!(__temp$(, $feed1)?)).into_iter();
-        $(
-            let chain_name!(__temp$(, $mod2)?) = $item2;
-            let $var = $var.chain(chain_skip!(__temp$(, $feed2)?));
-        )+
-    };
+pub(crate) fn i2osp_2_array<L: ArrayLength<u8> + IsLess<U256>>(
```

Comment:
Why is this named `i2osp_2_array`? Do you perhaps just mean `i2osp_array`?

Reply:
As above.